WSFrontend/org.openl.rules.ruleservice.ws.storelogdata.hive/src/org/openl/rules/ruleservice/storelogdata/hive/HiveEntityDao.java

```diff
@@ -41,12 +44,16 @@
         ZonedDateTime.class,
         Date.class));
 
-    public HiveEntityDao(Connection connection, Class<?> entityClass) throws SQLException,
-                                                                      UnsupportedFieldTypeException {
+    public HiveEntityDao(Class<?> entityClass) throws UnsupportedFieldTypeException {
         checkTypes(entityClass);
-        preparedStatement = new HiveStatementBuilder(connection, entityClass).buildInsertStatement();
+        sqlQuery = new HiveStatementBuilder(entityClass).buildQuery();
+        sortedPartitionedFields = Arrays.stream(entityClass.getDeclaredFields())
+                .filter(f -> !f.isSynthetic() && f.isAnnotationPresent(Partition.class))
+                .sorted(Comparator.comparingInt(f -> f.getAnnotation(Partition.class).value()))
+                .peek(f -> f.setAccessible(true))

```

Comment:
The "peek" operation marked as _mainly for debugging purposes_. Is there **peek** used instead of **map** for a legal reason?

Reply:
Mainly, but not only for. Yes, the functionality of _peek_ is exactly what should be done with elements. The _map_ would look like  
`.map(f -> {f.setAccessible(true); return f;})`